frontend/context/BalancesAndRefillRequirementsProvider.tsx

```diff
@@ -182,17 +205,33 @@ export const BalancesAndRefillRequirementsProvider = ({
     chainId,
     balancesAndFundingRequirements,
   ]);
+
   const getRefillRequirementsOf = useCallback(
     <T extends AddressBalanceRecord | MasterSafeBalanceRecord>(
       chainId: EvmChainId,
     ): Optional<T> => {
-      if (isBalancesAndFundingRequirementsLoading) return;
-      if (!balancesAndFundingRequirements) return;
+      if (isBalancesAndFundingRequirementsLoadingForAllServices) return;
+      if (!balancesAndFundingRequirementsForAllServices) return;
 
       const chain = asMiddlewareChain(chainId);
-      return balancesAndFundingRequirements.refill_requirements[chain] as T;
+      const serviceIdOfService = services?.find(
+        (s) => s.home_chain === chain,
```

Comment:
let's also add service_public_id to the filter?

Reply:
or do you think we should do a findAll? and then sum all the requirements. 
(shouldn't be needed right now, but when we start supporting multiple agents per chain)